File: src/api/schemas/location.py
from typing import Optional
from pydantic import BaseModel, Field

class LocationBase(BaseModel):
    location_name: str = Field(..., min_length=1, max_length=200)
    address: Optional[str] = Field(None, max_length=255)
    city: Optional[str] = Field(None, max_length=100)
    state_province: Optional[str] = Field(None, max_length=50)
    postal_code: Optional[str] = Field(None, max_length=20)
    notes: Optional[str] = None

class LocationCreate(LocationBase):
    pass

class LocationUpdate(BaseModel):
    location_name: Optional[str] = Field(None, min_length=1, max_length=200)
    address: Optional[str] = Field(None, max_length=255)
    city: Optional[str] = Field(None, max_length=100)
    state_province: Optional[str] = Field(None, max_length=50)
    postal_code: Optional[str] = Field(None, max_length=20)
    notes: Optional[str] = None
    active_status: Optional[bool] = None

class LocationResponse(LocationBase):
    id: int
    active_status: bool

    class Config:
        from_attributes = True

class LocationSearchRequest(BaseModel):
    name: Optional[str] = None
    city: Optional[str] = None
    active_only: bool = True

File: src/ui/src/types/location.ts
export interface Location {
  id: number
  location_name: string
  address?: string
  city?: string
  state_province?: string
  postal_code?: string
  active_status: boolean
  notes?: string
}

export interface CreateLocationRequest {
  location_name: string
  address?: string
  city?: string
  state_province?: string
  postal_code?: string
  notes?: string
}

export interface UpdateLocationRequest {
  location_name?: string
  address?: string
  city?: string
  state_province?: string
  postal_code?: string
  notes?: string
}

export interface LocationSearchRequest {
  name?: string
  city?: string
  active_only: boolean
}